src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
I think that `_fileSystem.ToShortRelativePath` will throw an exception on the file that doesn't exist